### ISSUE TITLE
How does allennlp support transfer learning ?

### ISSUE THREAD

**Issue by robbine** [allenai/allennlp#2360]
I trained a bert model using some Chinese article data with a Chinese dataset reader and a masked LM loss. Now my new task is a ner problem which I hope to introduce the transformer part from the bert model , Does allennlp support this kind of transfer learning ? Obviously ner problem requires a different dataset reader and crf loss.
torch provides load/save function like this
`checkpoint = torch.load(PATH)
model.load_state_dict(checkpoint['model_state_dict'])`
I wonder if I can load part of the model only, for example the transformer module from 'bert/best.th' model file, and continue on training ner model

**Comment by matt-gardner** [comment]
I would recommend using the [`pretrained` initializer](https://github.com/allenai/allennlp/blob/6bd59751e994f18251f69ed432c5f045991c90ed/allennlp/nn/initializers.py#L193-L233) (introduced in #2043).

**Comment by robbine** [comment]
Thanks ！
I guess all I need is to modify my training config file.
Does it mean to put the following part in the 'model' section ?
`"initializer" : ["linear_1.weight|linear_2.weight",
           {
               "type": "pretrained",
               "weights_file_path": "best.th",
               "parameter_name_overrides": {
                   "linear_2.weight": "linear_3.weight"
               }
           }
       ]`

**Comment by matt-gardner** [comment]
I suggest you look through the documentation and/or code to figure out what exactly those parameters do, then try to modify them for your particular use case.  Those values certainly won't work for you without modification.

**Comment by robbine** [comment]
Sure, Thanks ! This method works!
One more advice would be to add a parameter to specify which device the pretrained model is loaded .
Or else an exception is raised if cuda device is busy.
torch.load has a map_location parameter to do this.

File "/anaconda3/lib/python3.7/site-packages/allennlp/nn/initializers.py", line 238, in __init__
    self.weights: Dict[str, torch.Tensor] = torch.load(weights_file_path)

**Comment by matt-gardner** [comment]
I think moving the model to the gpu happens after the model is loaded, so I'm surprised this is an issue. Can you give a full stack trace?

**Comment by robbine** [comment]
The full stack trace is lost. 
I check the source code in  /anaconda3/lib/python3.7/site-packages/torch/serialization.py
and find these two lines.
register_package(10, _cpu_tag, _cpu_deserialize)
register_package(20, _cuda_tag, _cuda_deserialize)
It seems that cuda has a higher priority than cpu, so the exception is raised.

**Comment by matt-gardner** [comment]
This is not enough information for me to understand the problem, so I can't really help you fix it.  I'm going to close this issue, as your original question has been answered.  If you're still having trouble with loading the model, please open a new issue and follow the issue template, giving us enough information that we can help you solve the problem.
